test/NuGet.Core.FuncTests/Dotnet.Integration.Test/MsbuildIntegrationTestFixture.cs

```diff
@@ -451,5 +477,92 @@ private static void DeleteDirectory(string path)
+            string[] fullnames = new string[3];
+            for(int i = 0; i < filenames.Length; i++) 
+            {
+                fullnames[i] = prefix + Path.DirectorySeparatorChar + filenames[i];
+            }
+            TempPatchingDepsJsonForNewlyAddedDlls(fullnames);
+
+        }
+        private void TempCopyNewlyAddedDlls(string[] dlls, string pathCopyFrom, string PathCopyTo)
+        {
+            foreach (var dll in dlls) 
+            {
+                var copyFromFullName = Path.Combine(pathCopyFrom, dll);
+                var copyToFullName = Path.Combine(PathCopyTo, dll);
+                File.Copy(copyFromFullName, copyToFullName);
+            }
+        }
+        private void TempPatchingDepsJsonForNewlyAddedDlls(string[] filePaths)
+        {
+            string nugetBuildTasksName = "NuGet.Build.Tasks/5.3.0-rtm.6251";
+            foreach (string file in filePaths)
+            {
+                JObject JsonFile = GetJson(file);
+
+                JObject targets = null;
+                targets = JsonFile.GetJObjectProperty<JObject>("targets");
+
+                JObject netcoreapp50 = null;
+                netcoreapp50 = targets.GetJObjectProperty<JObject>(".NETCoreApp,Version=v5.0");
+
+                JObject NuGet_Build_Tasks = null;
+                NuGet_Build_Tasks = netcoreapp50.GetJObjectProperty<JObject>(nugetBuildTasksName);
+
+                JObject runtime = null;
+                runtime = NuGet_Build_Tasks.GetJObjectProperty<JObject>("runtime");
+
+                var jproperty = new JProperty("lib/netstandard2.1/System.Security.Cryptography.Pkcs.dll",
+                            new JObject
+                                {
+                                    new JProperty("assemblyVersion","4.0.4.0"),
+                                    new JProperty("fileVersion", "5.0.19.47301"),
+                                }
+                            );
+                runtime.Add(jproperty);
+                NuGet_Build_Tasks["runtime"] = runtime;
+                netcoreapp50[nugetBuildTasksName] = NuGet_Build_Tasks;
+                targets[".NETCoreApp,Version=v5.0"] = netcoreapp50;
+                JsonFile["targets"] = targets;
+                SaveJson(JsonFile, file);
+            }
+        }
+        private JObject GetJson(string jsonFilePath)
+        {
+            try
+            {
+                return FileUtility.SafeRead(jsonFilePath, (stream, filePath) =>
+                {
+                    using (var reader = new StreamReader(stream))
+                    {
+                        return JObject.Parse(reader.ReadToEnd());
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException(
+                    string.Format(jsonFilePath, ex.Message), ex);
```

Comment:
This makes no sense.  `ex.Message` will not be included in the exception message.  Please read [`string.Format(…)`](https://docs.microsoft.com/en-us/dotnet/api/system.string.format?view=netframework-4.8#System_String_Format_System_String_System_Object_) and verify your expectations.